src/Workspaces/Core/Portable/EmbeddedLanguages/RegularExpressions/RegexParser.cs

```diff
@@ -253,47 +257,107 @@ private RegexExpressionNode ParseAlternatingSequencesWorker(bool consumeClosePar
 
         private RegexSequenceNode ParseSequence(bool consumeCloseParen)
         {
-            var list = ArrayBuilder<RegexExpressionNode>.GetInstance();
-
+            var builder = ArrayBuilder<RegexExpressionNode>.GetInstance();
             while (ShouldConsumeSequenceElement(consumeCloseParen))
             {
-                var last = list.Count == 0 ? null : list.Last();
-                list.Add(ParsePrimaryExpressionAndQuantifiers(last));
-
-                TryMergeLastTwoNodes(list);
```

Comment:
`O(n^2)` for memory and cpu.  If you had a long sequence of text nodes (not at all hard if you just have `abcdefgh...`), then each text node would concat with the previous, recopying all of the previous node's contents and adding in the new text.

New merging algorithm is linear.  

Note: i gave some thought if we could make it so we can avoid merging the text nodes in the first place and just produce merged text tokens, but it proved challenging.